pkgs/by-name/co/couchbase-shell/package.nix

```
@@ -0,0 +1,63 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  openssl,
+  darwin,
+  testers,
+  nix-update-script,
+  couchbase-shell,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "couchbase-shell";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "couchbaselabs";
+    repo = "couchbase-shell";
+    rev = "v${version}";
+    hash = "sha256-ksAyi7yMz56de1lA2LYVNdsn02GNrcJVoRLcK1zFppE=";
+  };
+
+  cargoHash = "sha256-efy1M4Q9dBfEq0YYUKn4y1Rz/dPbBYLapcPXJLI9X+Q=";
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs =
+    [
+      openssl
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin (
+      with darwin.apple_sdk.frameworks;
+      [
+        SystemConfiguration
+        CoreServices
+      ]
+    );
+
+  # tests need couchbase server
+  doCheck = false;
+
+  passthru = {
+    tests.version = testers.testVersion {
+      package = couchbase-shell;
+    };
+    updateScript = nix-update-script { };
+  };
+
+  meta = {
+    description = "Shell for Couchbase Server and Cloud";
+    homepage = "https://couchbase.sh/";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ petrkozorezov ];
+    mainProgram = "cbsh";
+    platforms = [
+      "x86_64-linux"
+      "aarch64-darwin"
+    ];
```

Seems to work fine on aarch64-linux as well, presumably it works on linux & darwin on any architecture.
```suggestion
    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```